algoliasearch/src/main/java/com/algolia/search/saas/BaseIndex.java

```diff
@@ -310,6 +310,7 @@ protected JSONObject deleteObjects(List<String> objects) throws AlgoliaException
     protected void deleteByQuery(Query query) throws AlgoliaException {
         query.setAttributesToRetrieve("objectID");
         query.setHitsPerPage(100);
+        query.enableDistinct(false);
```

Comment:
You should never modify the client's query behind the scenes. If you need to change parameters, you should clone it. (Same applies to `Index.java#427`.)